database/redis/metric.go

```diff
@@ -136,40 +137,87 @@ func (connector *DbConnector) SaveMetrics(metrics map[string]*moira.MatchedMetri
 			}
 
 			var metricEventsChannel = metricEventsChannels[rand.Intn(len(metricEventsChannels))]
-			if err = c.Publish(ctx, metricEventsChannel, event).Err(); err != nil {
-				return err
-			}
+			pipe.SAdd(ctx, metricEventsChannel, event)
 		}
 	}
+
+	if _, err = pipe.Exec(ctx); err != nil {
+		connector.logger.Errorf("Sending metric event error. Error: %v;", err)
+		return err
+	}
+
 	return nil
 }
 
 // SubscribeMetricEvents creates subscription for new metrics and return channel for this events
-func (connector *DbConnector) SubscribeMetricEvents(tomb *tomb.Tomb) (<-chan *moira.MetricEvent, error) {
-	metricsChannel := make(chan *moira.MetricEvent, pubSubWorkerChannelSize)
-	dataChannel, err := connector.manageSubscriptions(tomb, metricEventsChannels)
-	if err != nil {
+func (connector *DbConnector) SubscribeMetricEvents(tomb *tomb.Tomb, params *moira.SubscribeMetricEventsParams) (<-chan *moira.MetricEvent, error) {
+	responesChannel := make(chan string, pubSubWorkerChannelSize)
+	metricChannel := make(chan *moira.MetricEvent, pubSubWorkerChannelSize)
```

Comment:
How do you think, maybe rename `pubSubWorkerChannelSize` and `pubSubWorkerChannelSize`, because we don't use pub/sub model anymore? Or now we continue to use pub/sub model but with Queue as implementation?

Reply:
Agree